cache/compressed_secondary_cache.cc

```diff
@@ -222,14 +228,22 @@ Status CompressedSecondaryCache::InsertInternal(
 
   PERF_COUNTER_ADD(compressed_sec_cache_insert_real_count, 1);
   if (cache_options_.enable_custom_split_merge) {
-    size_t charge{0};
-    CacheValueChunk* value_chunks_head =
-        SplitValueIntoChunks(val, cache_options_.compression_type, charge);
-    return cache_->Insert(key, value_chunks_head, internal_helper, charge);
+    size_t split_charge{0};
+    CacheValueChunk* value_chunks_head = SplitValueIntoChunks(
+        val, cache_options_.compression_type, split_charge);
+    return cache_->Insert(key, value_chunks_head, internal_helper,
+                          split_charge);
   } else {
+#ifdef ROCKSDB_MALLOC_USABLE_SIZE
+    charge = malloc_usable_size(ptr.get());
+#else
+    charge = total_size;
+#endif
+    EncodeFixed64(payload - sizeof(uint64_t), data_size);
```

Comment:
Although an extra 8 bytes is relatively small, it seems like every little bit of space helps when talking about compressed cache entries. Perhaps we can varint encode how much to subtract from the charge to get the data size? I know there's a recursive dependency where the value encoded might depend on the size of the encoded value, but I believe there is a clever solution: if the encoded value is going to be more than one byte, that means you have internal fragmentation that you can eat into, beyond your requested allocation size. (optimize_filters_for_memory already makes use of buffers beyond the requested size up to the reported usable size)

Or simply varint encoding the data size would reap almost all of the gain (within about half a byte), and I believe avoid the hack around the recursive dependency.